pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -317,34 +307,66 @@ func (ctrl *csiSnapshotSideCarController) createSnapshotOperation(content *crdv1
 		return content, nil
 	}
 
+	return ctrl.createSnapshotWrapper(content)
+}
+
+// The function wraps the snapshot creation.
+func (ctrl *csiSnapshotSideCarController) createSnapshotWrapper(content *crdv1.VolumeSnapshotContent) (*crdv1.VolumeSnapshotContent, error) {
+	klog.Infof("createSnapshotWrapper: Creating snapshot for content %s through the plugin ...", content.Name)
+
```

Comment:
IMO, checking the errors in the status is needed here to avoid unnecessary calls and annotation updates, if snapshot creation has failed permanently previously, all this logic is not needed

Reply:
Sure